servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/IoExecutor.java

```diff
@@ -15,15 +15,15 @@
  */
 package io.servicetalk.transport.api;
 
+import io.servicetalk.concurrent.Executor;
 import io.servicetalk.concurrent.api.ListenableAsyncCloseable;
 
-import java.util.concurrent.Executor;
 import java.util.function.BooleanSupplier;
 
 /**
  * {@link Executor} that handles IO.
  */
-public interface IoExecutor extends ListenableAsyncCloseable {
+public interface IoExecutor extends Executor, ListenableAsyncCloseable {
```

Comment:
1. We can probably remove `Executor` declaration from `AbstractNettyIoExecutor` now.
2. This is ok change for the main branch (0.43 version). If you are going to cherry pick this into 0.41 and 0.42, consider adding default implementations that throw `UnsupportedOperationException`. As you mentioned in the commit description, nobody will probably be broken either way. Having default impls will help to avoid any ABI issues.

Reply:
1. `AbstractNettyIoExecutor` is implementing the `io.servicetalk.concurrent.api.Executor` sub-interface so still needs to declare it. For the same reason `NettyIoExecutor` still needs to provide the `asExecutor()` for returning the casted instance. We could have `NettyIoExecutor` extend `io.servicetalk.concurrent.api.Executor` perhaps.
2. Providing defaults makes sense for 0.42 but backporting to 0.41 would be too much work for the benefit I suspect.

Reply:
1. 👍 
2. The change of this PR is small and defaults will be the same for 0.42 and 0.41. Why do you expect 0.41 to be much different?

Reply:
> The change of this PR is small and defaults will be the same for 0.42 and 0.41. Why do you expect 0.41 to be much different?

Sorry, I was thinking of the follow-on bug that will use this new feature. I have no objections to back-porting this specific change to both 0.41 and 0.42